lib/zlib/inflate.js

```diff
@@ -207,7 +207,7 @@ function inflateReset(strm) {
   var state;
 
   if (!strm || !strm.state) { return Z_STREAM_ERROR; }
-  state = strm.state;
+  state = new InflateState();
```

Comment:
I guess, that's wrong. everything in `/zlib` folder is ported from upstream, and does exactly what it should. Problems on upper wrapped should not affect this code.